packages/safe-service-client/src/SafeTransactionService.ts

```diff
@@ -0,0 +1,78 @@
+import { SafeSignature, SafeTransactionData } from '@gnosis.pm/safe-core-sdk-types'
+import {
+  MasterCopyResponse,
+  OwnerResponse,
+  SafeBalanceResponse,
+  SafeBalanceUsdResponse,
+  SafeCollectibleResponse,
+  SafeCreationInfoResponse,
+  SafeDelegate,
+  SafeDelegateDelete,
+  SafeDelegateListResponse,
+  SafeInfoResponse,
+  SafeModuleTransactionListResponse,
+  SafeMultisigConfirmationListResponse,
+  SafeMultisigTransactionEstimate,
+  SafeMultisigTransactionEstimateResponse,
+  SafeMultisigTransactionListResponse,
+  SafeMultisigTransactionResponse,
+  SafeServiceInfoResponse,
+  SignatureResponse,
+  TokenInfoListResponse,
+  TokenInfoResponse,
+  TransferListResponse
+} from './types/safeTransactionServiceTypes'
+
+interface SafeTransactionService {
+  // About
+  getServiceInfo(): Promise<SafeServiceInfoResponse>
+  getServiceMasterCopiesInfo(): Promise<MasterCopyResponse>
+
+  // Data decoder
+  decodeData(data: string): Promise<any>
+
+  // Owners
+  getSafesByOwner(ownerAddress: string): Promise<OwnerResponse>
+
+  // Multisig transactions
+  getTransaction(safeTxHash: string): Promise<SafeMultisigTransactionResponse>
+  getTransactionConfirmations(safeTxHash: string): Promise<SafeMultisigConfirmationListResponse>
+  confirmTransaction(safeTxHash: string, signature: string): Promise<SignatureResponse>
+
+  // Safes
+  getSafeInfo(safeAddress: string): Promise<SafeInfoResponse>
+  getSafeDelegates(safeAddress: string): Promise<SafeDelegateListResponse>
+  addSafeDelegate(safeAddress: string, delegate: SafeDelegate): Promise<any>
+  removeSafeDelegate(safeAddress: string, delegate: SafeDelegateDelete): Promise<any>
+
+  // Transactions
+  getSafeCreationInfo(safeAddress: string): Promise<SafeCreationInfoResponse>
+  estimateSafeTransaction(
+    safeAddress: string,
+    safeTransaction: SafeMultisigTransactionEstimate
+  ): Promise<SafeMultisigTransactionEstimateResponse>
+  proposeTransaction(
+    safeAddress: string,
+    transaction: SafeTransactionData,
+    transactionHash: string,
```

Comment:
I was confused by transactionHash (what tx hash is it?) naming, maybe it's better to use `contractTransactionHash` as it's named in the request body or `safeTxHash`